CMakePresets.json

```diff
@@ -127,6 +140,102 @@
                 "CXX": "clazy",
                 "CCACHE_DISABLE": "ON"
             }
+        },
+        {
+            "name": "ci-dev-client-only-qt5",
+            "displayName": "ci-dev-client-only-qt5",
+            "description": "Qt5 client-only build",
+            "binaryDir": "${sourceDir}/build-ci-dev-client-only-qt5",
+            "generator": "Ninja",
```

Comment:
I was just reading the [cmake-presets docs](https://cmake.org/cmake/help/latest/manual/cmake-presets.7.html#macro-expansion) -- they are _clever_ sometimes.

> All macros are evaluated in the context of the preset being used, even if the macro is in a field that was inherited from another preset. For example, if the `Base` preset sets variable `PRESET_NAME` to `${presetName}`, and the Derived preset inherits from `Base`, `PRESET_NAME` will be set to `Derived`.

AFAICT that means you could just set `"binaryDir"` to `"${sourceDir}/build-${presetName}"` once in the `base` config, and you're done.

(Tho that might actually make things harder for any `testPresets` configs I might propose, since it would mean there would have to be a 1:1 correlation of `buildPresets` names and `testPresets` names ...)

Reply:
> (Tho that might actually make things harder for any `testPresets` configs I might propose, since it would mean there would have to be a 1:1 correlation of `buildPresets` names and `testPresets` names ...)

Actually, I might be wrong. They say macros expand in the current context, but also that `testPresets` and `buildPresets` both inherit their build dir from the `configurationPreset` --- so that should mean the `binaryDir` name expansion happens with the name of the `configurationPreset` as `presetName`? Hmm. I'll have to experiment a bit.